packages/ses/src/error/assert.js

```diff
@@ -401,6 +401,45 @@ const makeAssert = (optRaise = undefined, unredacted = false) => {
   const assertString = (specimen, optDetails) =>
     assertTypeof(specimen, 'string', optDetails);
 
+  /** @type {NotThrows} */
+  const notThrows = action => {
+    try {
+      return action();
+    } catch (reason) {
+      // Abandons/terminates the unit of computation this `assert` instance
+      // is supposed to. `zcf.assert.notThrows(f)` will immediately shut down
+      // the contract if `f()` throws.
+      throw fail(details`Failure was not an option: ${reason}`);
+    }
+  };
+
+  /** @type {Atomic} */
+  const atomic = action => {
```

Comment:
This does not seem like atomicity. It only gets near atomicity because of a lot of other infrastructure that this library cannot and should not count on (e.g., if I fail the cat I'm assured the nothing got out). There's no need to try to call it "atomic". This seems like "total", but it's not that either, similarly since things can get partially done and we just promise to destroy all the evidence of that :)